### PR TITLE
Put GeoPackage at first place in the menu

### PR DESCRIPTION
**Pull request by jachym**
## Description
This PR puts OGC GeoPackage at first place in the "Create new vector layer" menu entry.

geopackagefirst

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nyalldawson** [comment]
Can you add a "[needs-docs]" tag to the commit message? The change in shortcut behavior will need to be updated in the qgis documentation

**Comment by DelazJ** [comment]
Question: does this also fix the toolbar?

**Comment by luipir** [comment]
@DelazJ should the tool bar buttons be substituted by the single add layer button ? IMHO the buttons remain just for a transition or nostalgic phase

**Comment by DelazJ** [comment]
These are buttons to **create** a new layer not to add a layer...
Do you mean that layer creation buttons are also meant to move to the unified **add** layer dialog?
My comment was to provide same shortcut/sort using the menu Layer -> Create layer than using the "group-menu" button from layer toolbar

**Comment by luipir** [comment]
@DelazJ  about creation with single add button at least gpkg and splite, I don't remember if it's possibile also for memory and shape.
@elpaso any opinion about menu layer?

**Comment by DelazJ** [comment]
Not sure I follow you. In case, I am/was referring to these buttons (reorder the sequence and link shortcut to GPKG)
image

**Comment by jachym** [comment]
Oh, I was not aware of the toolbar button - ok, will fix it also for the toolbar too.

Otherwise no objections?

**Comment by luipir** [comment]
@jachym go ahead :)

**Comment by jachym** [comment]
So, to me it seems, this menu does not need any change
qgis

**Comment by jachym** [comment]
@luipir Order of "new file" buttons in the toolbar changed as well
@nyalldawson Tags add

Anything else?

**Comment by rduivenvoorde** [comment]
@jachym well, while you are on it ;-)
What about the Save Vector Layers as dialog Format dropdown?
(from layermanager, right click menu item Save As):

selection_131

I think that is a good candidate too?

**Comment by jachym** [comment]
@rduivenvoorde and @luipir I've add another commit with some other dialog changes eb964ef930a76020395a92dcc48189d256b815df can you guys review?

image

**Comment by luipir** [comment]
I can't ckeck for all conditions, but seems it's the correct way to get rid of shapefile :)... sometimes so little PRs can have a great effect. Great :)

**Comment by jachym** [comment]
@luipir or great side effect - any hint, what to do with the travis check? apparently, it's complainig about api change

```
SIP file is not up to date
core/qgsvectorfilewriter.sip
```

**Comment by Gustry** [comment]
@jachym Did you run the scripts/prepare-commit.sh before the commit?
It will run "sipify" for you, or you can run it manually: https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html#id30

**Comment by jachym** [comment]
@Gustry well I run `scripts/sipify.pl` manulally now - 

` scripts/prepare-commit.sh`
gave me error: 
```
Running spell check on files: python/plugins/db_manager/dlg_export_vector.py resources/qgis-resource-metadata.xml scripts/random_vector.py src/app/qgsattributeactiondialog.cpp src/core/processing/qgsprocessingutils.cpp src/core/qgsvectorfilewriter.cpp src/core/qgsvectorfilewriter.h src/gui/ogr/qgsvectorlayersaveasdialog.cpp src/plugins/geometry_checker/ui/qgsgeometrycheckerresulttab.cpp src/plugins/geometry_checker/ui/qgsgeometrycheckersetuptab.cpp src/providers/ogr/qgsogrprovider.cpp 
couldn't execute "ag": no such file or directory
    while executing
"spawn -noecho ag --noaffinity --all-text --nopager --color-match {30;43} --numbers --nomultiline --ignore-case -p /home/jachym/src/cpp/QGIS/scripts/sp..."
    ("eval" body line 1)
    invoked from within
"eval [list spawn -noecho] $argv"
    invoked from within
"if {[string compare [lindex $argv 0] "-p"] == 0} {
    # pipeline
    set stty_init "-echo"
    eval [list spawn -noecho] [lrange $argv 1 end]
    clo..."
    (file "/usr/bin/unbuffer" line 13)
couldn't execute "ag": no such file or directory
```
any hint?

**Comment by m-kuhn** [comment]
Some tests apparently assume shapefiles as default.
Switching to gpkg here should be trivial:
https://github.com/qgis/QGIS/blob/master/tests/src/analysis/testqgsprocessing.cpp#L1212-L1214
Not sure exactly what happens here though:
https://github.com/qgis/QGIS/blob/master/tests/src/core/testqgsmaprendererjob.cpp#L179-L182

**Comment by jachym** [comment]
neither do I :-(

**Comment by jachym** [comment]
66 tests failing - but they do not seem to be by far related to "shp" issue :-/

**Comment by nyalldawson** [comment]
I'm -1 on the API changes. I think changing user facing settings is fine, but changing default API methods to use gpkg is premature and may have many unintentional side effects.

File: resources/qgis-resource-metadata.xml [hunk 1]
```diff
@@ -52,7 +52,7 @@
     <links>
         <link name="geonode:roads" type="OGC:WMS" description="my GeoNode road layer" url="http://example.org/wms"/>
         <link name="geonode:roads" type="OGC:WFS" description="my GeoNode road layer" url="http://example.org/wfs"/>
-        <link name="roads" type="WWW:LINK" description="full dataset download" url="http://example.org/roads.tgz" format="ESRI Shapefile" mimeType="application/gzip" size="283676"/>
+        <link name="roads" type="WWW:LINK" description="full dataset download" url="http://example.org/roads.tgz" format="GPKG" mimeType="application/gzip" size="283676"/>
```
Comment (nyalldawson) [inline resources/qgis-resource-metadata.xml]: This change should be removed

File: scripts/random_vector.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def error(msg):
 
 (minx, miny, maxx, maxy) = map(float, options.extent.split(","))
 
-driverName = "ESRI Shapefile"
+driverName = "GPKG"
```
Comment (nyalldawson) [inline scripts/random_vector.py]: Revert this change too - it's a purely test based tool, switching to gpkg has no benefit here

File: resources/qgis-resource-metadata.xml (hunk 1 above)
Comment (jachym) [reply resources/qgis-resource-metadata.xml]: you are right, changed back

File: scripts/random_vector.py (hunk 2 above)
Comment (jachym) [reply scripts/random_vector.py]: changed back

**Comment by jachym** [comment]
Thank you  @nyalldawson for review, any more comments?

File: src/app/qgsattributeactiondialog.cpp [hunk 3]
```diff
@@ -306,7 +306,7 @@ void QgsAttributeActionDialog::addDefaultActions()
 {
   int pos = 0;
   insertRow( pos++, QgsAction::Generic, tr( "Echo attribute's value" ), QStringLiteral( "echo \"[% \"MY_FIELD\" %]\"" ), QLatin1String( "" ), true, tr( "Attribute Value" ), QSet<QString>() << QStringLiteral( "Field" ), QString() );
-  insertRow( pos++, QgsAction::Generic, tr( "Run an application" ), QStringLiteral( "ogr2ogr -f \"ESRI Shapefile\" \"[% \"OUTPUT_PATH\" %]\" \"[% \"INPUT_FILE\" %]\"" ), QLatin1String( "" ), true, tr( "Run application" ), QSet<QString>() << QStringLiteral( "Feature" ) << QStringLiteral( "Canvas" ), QString() );
+  insertRow( pos++, QgsAction::Generic, tr( "Run an application" ), QStringLiteral( "ogr2ogr -f \"DPKG\" \"[% \"OUTPUT_PATH\" %]\" \"[% \"INPUT_FILE\" %]\"" ), QLatin1String( "" ), true, tr( "Run application" ), QSet<QString>() << QStringLiteral( "Feature" ) << QStringLiteral( "Canvas" ), QString() );
```
Comment (luipir) [inline src/app/qgsattributeactiondialog.cpp]: Should be GPKG and not DPKG

**Review by luipir — CHANGES_REQUESTED** [review]
Should be GPKG and not DPKG

**Comment by luipir** [comment]
@nyalldawson at what api change are you referring... btw I feel more confortable to this kind of api change than more new features. IMHO This will force us to fix unconsistent use fo gpkg.
Just my 2c

File: src/app/qgsattributeactiondialog.cpp (hunk 3 above)
Comment (jachym) [reply src/app/qgsattributeactiondialog.cpp]: oh, thanks, fixed

**Comment by jachym** [comment]
Horray, travis passed

**Comment by nyalldawson** [comment]
Looks good to me.

**Comment by jachym** [comment]
@luipir are you ok with current pull request? can you guys push, to push this to master?

**Comment by luipir** [comment]
go ahead :)

**Comment by nyalldawson** [comment]
I've had to revert the changes to testqgsmaprendererjob - they've been causing hangs on Travis (see e.g. https://travis-ci.org/qgis/QGIS/jobs/290839698 )

**Comment by jachym** [comment]
OMFG, thanks a lot @m-kuhn !!
